vaadin-checkbox-flow-parent/vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/Checkbox.java

```diff
@@ -47,8 +63,7 @@ public class Checkbox extends GeneratedVaadinCheckbox<Checkbox, Boolean>
      * Default constructor.
      */
     public Checkbox() {
-        // initial value, default value, accept null value
-        super(false, false, false, true);
+        super("checked", false, false);
```

Comment:
There is a subtle difference here compared to the value initialization from the generated class. Since the initial value is not null (`false`), and in case the `checked` property value is null, the generated class would call effectively call `setPresentationValue(false)`, which would initialize the `checked` property with `false`.

Now after the change, the `checked` property is `null` after initialization. The only difference I can see is an extra sync to the server, probably because the web component initializes the value with `false`:

<img width="1446" alt="Bildschirmfoto 2022-12-12 um 11 47 01" src="https://user-images.githubusercontent.com/357820/207028673-7d1630dd-a866-4e9a-a1da-a6beaef7bbe0.png">

If we want to restore the behavior, we would have to add something like:
```java
if (getElement.getProperty("checked") == null) {
    setPresentationValue(false);
}
```
The condition is necessary in order to avoid overwriting any initial value set in a template.

Reply:
Added the check. Should I add some unit test to cover this change? Existing tests didn't catch this.
There is `InjectedCheckboxIT` but it doesn't test `checked` attribute being set in a template.

Reply:
One case (checkbox element has initial value) is tested here:
https://github.com/vaadin/flow-components/blob/8bbbb99f5005284b6288bfa1c3bcafe55baf1cf2/vaadin-checkbox-flow-parent/vaadin-checkbox-flow/src/test/java/com/vaadin/flow/component/checkbox/tests/CheckboxUnitTest.java#L90-L110

We could add an additional test where we just create a new checkbox instance and verify that the property is initialized with `false` by default.

Reply:
Thanks, makes sense. I will do that.